app/views/foreman_kernel_care/job_templates/update_kernel.erb

```diff
@@ -3,13 +3,19 @@ kind: job_template
 name: Update kernel
 model: JobTemplate
 job_category: Commands
-description_format: "Run yum update kernel"
+description_format: "Update kernel"
 provider_type: SSH
 feature: update_kernel
 %>
-<%
-  unless @host.operatingsystem.family == 'Redhat'
-    render_error(N_('Unsupported operating system for this host.'))
-  end
-%>
-yum update -y kernel*
+
+<%= snippet 'pkg_manager' %>
+
+<% if @host.operatingsystem.family == 'Redhat' -%>
+KERNEL_NAME="kernel*"
+<% elsif @host.operatingsystem.family == 'Debian' -%>
+KERNEL_NAME="linux-image-generic"
+<% else -%>
+<%= render_error(N_('Unsupported operating system for this host.')) %>
+<% end %>
+
+$PKG_MANAGER_UPGRADE $KERNEL_NAME && /usr/bin/kcarectl --update
```

Comment:
I know I am late to the party, but I don't really understand why the kernel should be updated by the package manager. I thought that was supposed to be done be the `/usr/bin/kcarectl --update` command. Also, during testing this throws an error on my host. 
```
...
Extracting templates from packages: 96%
Extracting templates from packages: 100%
  17: Preconfiguring packages ...
  18: 30 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
  19: Need to get 0 B/76.4 MB of archives.
  20: After this operation, 318 MB of additional disk space will be used.
  21: dpkg: error: need an action option
  22: 
  23: Type dpkg --help for help about installing and deinstalling packages [*];
  24: Use 'apt' or 'aptitude' for user-friendly package management;
  25: Type dpkg -Dhelp for a list of dpkg debug flag values;
  26: Type dpkg --force-help for a list of forcing options;
  27: Type dpkg-deb --help for help about manipulating *.deb files;
  28:
  29: Options marked [*] produce a lot of output - pipe it through 'less' or 'more' !
  30: No updates performed. See /var/log/rhsm/rhsm.log for more information.
  31: E: Sub-process /usr/bin/dpkg returned an error code (2)
  32: Exit status: 100
```

Reply:
The main idea was to update kernel package to the latest possible version because kernelcare or kpatch updates only running kernel. It was the idea of kernelcare management. Personally, I don't see any point in this update. So maybe we can remove it in the future version.